src/reducers/breakpoints.js

```diff
@@ -118,12 +114,17 @@ function addXHRBreakpoint(state, action) {
   );
 
   if (existingBreakpointIndex === -1) {
-    return state.set("xhrBreakpoints", xhrBreakpoints.push(breakpoint));
+    // TODO make this immutable?
+    return {
```

Comment:
it is okay to have mutable state as long as you copy it and ensure that it does not change. We ensure this with source records via flow 

e.g. `type Source = {| +a: 2 |}; let s: Source = {a: 2}`. Now `a` is read only :)